[4.0] Display of horizontal mod_articles_news module

drmenzelit opened the pull request:
Pull Request for Issue #30426 .

### Summary of Changes
Added a new scss file to collect all stuff related with modules
Added a display: grid (with flex fallback) to the class .mod-articlesnews-horizontal

### Testing Instructions
Install PR and run npm ci or compile template scss.
Create some articles and a newsflash module with layout horizontal. The width is optimized for 3 articles side by side.

### Actual result BEFORE applying this Pull Request
You see all articles as list one below the other

![Screenshot_2020-08-31 Home-Newsflash](https://user-images.githubusercontent.com/9153168/91719522-c2dae000-eb95-11ea-8370-d4901b425472.png)


### Expected result AFTER applying this Pull Request
You will see the articles side by side
![Screenshot_2020-08-31 Home-Newsflash-horiz](https://user-images.githubusercontent.com/9153168/91719664-06354e80-eb96-11ea-87db-fe284f100ce2.png)
![Screenshot_2020-08-31 Home-Newsflash-horiz-2col](https://user-images.githubusercontent.com/9153168/91719672-09303f00-eb96-11ea-95d6-8e30cadee492.png)


### Documentation Changes Required

drmenzelit committed:
Merge remote-tracking branch 'joomla-origin/4.0-dev' into 4.0-dev

infograf768 committed:
[4.0] Fix Alert double error display in Installation

infograf768 committed:
seperating multiple messages with border top

Commit:
Increase minimum length

Commit:
Fix meter displaying complete message prematurely

Commit:
Add strenthmeter attribute to installation

Commit:
Change meter values to display green when password requirements are met

drmenzelit committed:
Merge remote-tracking branch 'joomla-origin/4.0-dev' into 4.0-dev

infograf768 committed:
[4.0] {Cassiopea] Implementing Password length/meter

infograf768 committed:
[4.0] Fix Fieldset description display (#30435)

infograf768 committed:
[4.0] Url Language Code is not a prefix (#30440)

* [4.0] URL LANGUAGE CODE is not a prefix

* modified desc as suggested

infograf768 committed:
[4.0] Implementing display of fieldset descriptions for fieldset
children

infograf768 committed:
Taking off description

infograf768 committed:
scss cs

infograf768 committed:
use @Quy suggestion

infograf768 committed:
file cs

dgrammatiko committed:
[4.0] Media field fix (#30455)

* fix

* typo, (well nothing new)

* Allow media fields in subforms

- Cleanup Media field
- Expose initilazation method of Bootstrpa Modlas to a Joomla.Bootstrap object
- Initilise Modlas on the lifecycle of the Custom Element

* Paleolithic js mode only in the legacy…

* CS

* Proper check

* Pffff

* Update joomla-field-media.w-c.es6.js

N6REJ committed:
[4.0] Convert icon used for jgrid defaults to fas fa- (#30464)

* make icon-circle use fa-circle

* allows for including icon- in call

N6REJ committed:
move fa-fw to last class to fit convention (#30469)

brianteeman committed:
[4.0] Long labels wrapping (#30474)

When we have fields displayed in columns with the label displayed above the input there is a width limit of 240px on the label.

This results in long labels wrapping when it is not needed as the column is wider than 240px.

In english we dont see this is many places but the easiest is in the __configure edit screen_  fields

This simple PR changes the label width to 100% when the fields are displayed in columns like this.

As this is an scss change you will need to `node build.js --compile-css` in order to test

### Before

### After

chmst committed:
Fix mod_articles_latest (#30459)

drmenzelit committed:
[4.0] Add a parameter for "back-to-top" button in Cassiopeia (#30441)

* [4.0] Add a parameter for back-to-top button in Cassiopeia

Added a parameter to show / hide the back-to-top button in Cassiopeia.
Currently the footer appears when a module has been published on position "footer". And the back-to-top only appears if the footer is visible. Since the footer is part of the grid, it is not easy to move the back-to-top outside the grid.

Now it is possible to decide to have a back-to-top or not. If yes, the footer will be rendered showing the button, independently if there is a module in the position footer.
It is also possible to show only modules (switch back-to-top off) or modules and back-to-top (switch on).

* Added smooth scrolling

Add smooth scrolling taking in account the preferences of the user (reduced motion).
Improve  language file.

* Correct height and position of back-to-top button

* Correct code style

* Correct code style 2

* Change smooth scrolling behavior to take in account reduced motion preferences

* Language file corrected, language string changed to match others

* Update language/en-GB/tpl_cassiopeia.ini

Link instead button

Co-authored-by: Brian Teeman <brian@teeman.net>

* Update templates/cassiopeia/scss/blocks/_global.scss

Comment for smooth scroll

Co-authored-by: Brian Teeman <brian@teeman.net>

Co-authored-by: Brian Teeman <brian@teeman.net>

N6REJ committed:
fix improper family name. (#30481)

brianteeman committed:
[4.0] update helptoc (#30490)

* [4.0] update helptoc

Removes the string for the expired cache page that no longer exists and removes it from the index displayed on the left hand side in the help page

* api

zero-24 committed:
[4.0] Add new permissions-policy to the HTTPHeaders Plugin (#30491)

* add permissions-policy

* add Permissions-Policy to the dropdown

* alpha order

Commit:
[4.0] Error when changing status of tagged content items (#30466)

* Correct column name

* Fix changing UCM state

brianteeman committed:
[4.0] br tag (#30503)

* [4.0] br tag

In Joomla 4 we use `<br>` not `<br />`

* bad grep

brianteeman committed:
[4.0] Remove old search component (#30506)

* [4.0] Remove old search component

The regular search component is not in J4

This pr removes reference to it in the help system
`administrator/index.php?option=com_admin&view=help`

* revert

brianteeman committed:
[4.0] Help Dashboard (#30508)

* [4.0] Help Dashboard

This PR makes multiple changes to the help dashboard. In order of importance they are.

- Created a new section "Start Here"
- Moved "Joomla Help" to the "Start Here" section to give it greater priority as it really is the first place you should look for help
- Removed target=_blank from "Joomla Help"
- Moved "Documentation Wiki" from "Resources" to the top of "Additional Help"

* deduplicate

* compass icon

drmenzelit committed:
First step to make newsflash horizontal

drmenzelit committed:
Merge branch '4.0-dev' of https://github.com/joomla/joomla-cms into 4.0-dev

drmenzelit committed:
Merge remote-tracking branch 'origin/4.0-dev' into 4.0-dev

drmenzelit committed:
Merge branch '4.0-dev-newsflash-horiz' of https://github.com/drmenzelit/joomla-cms.git; branch '4.0-dev' of https://github.com/joomla/joomla-cms into 4.0-dev-newsflash-horiz

dgrammatiko commented on templates/cassiopeia/scss/template.scss:
```diff
@@ -37,6 +37,7 @@
 @import "blocks/layout";
 @import "blocks/modals";
 @import "blocks/modifiers";
+@import "blocks/modules";
```
@drmenzelit fwiw you could save the `templates/cassiopeia/scss/blocks/_modules.scss` to a path `templates/cassiopeia/scss//mod_modules.scss`, remove the import from the template.scss and add a $wa->registerStylesheetAndUse(...) in the module layout

That way you're not exploding unnesseserily the template.css and also building things modular, which is both more maintainable, more performant and also the current best practice for anything web.

My 2c, good work

drmenzelit replied:
Oh, I didn't think about this possibility. Really good idea, each module will load its own css without bloating the general template :-)
Thank you for your 2c ;-)

drmenzelit replied:
@dgrammatiko if I use your idea, how can I have access to the variables defined for the template? It would not make sense to import _variables.scss into mod_modules.scss, then we will have the code twice...maybe I'm missing something...

dgrammatiko replied:
> how can I have access to the variables defined for the template?

These are css variables, right? If not they should be, it's 2020

dgrammatiko replied:
I did some modularization work some time ago, so maybe looking at the code on https://github.com/joomla/joomla-cms/pull/26077 will give you the answers you need

drmenzelit replied:
Sadly is Cassiopeia still using scss variables... we have a lot of work to do...

dgrammatiko replied:
Anyways the first ~10 lines from https://github.com/joomla/joomla-cms/pull/26077/files#diff-062253413198f3835ac02c79b6ba53d7 adjusted to Cassiopeia is all you need (importing the functions, mixins and vars doesn't explode the output)

Magnytu2 commented:
Good evening, it's great the choice of vertical or horizontal layout works.
But in the Main-top position, there is a two-column display. In the Top-a position, I have a display of my three articles. But if I choose 4 items, the fourth is on a second line.
I have the impression that there is a limit by the image size.
On the images, we can also see that they are cut which is a shame.
![Home-2](https://user-images.githubusercontent.com/26197929/91742679-e95d4300-ebb6-11ea-8ca5-bb733ddce676.jpg)

drmenzelit commented:
As I wrote in my explanation, the width of the grid is optimized for 3 articles side by side. You can overwrite this on your own css. The image has a width of 100% of the column and the height is set to 250px (can be changed too) to avoid having images of different sizes. If the user has always images with the same proportions, it should not be a problem. But I know a lot of users having different sizes for images and then the news will look very weird.

Magnytu2 commented:
Sorry, I did not understand this instruction.
It's already very good on three columns, but a pity that it is not proportional.
For the images, we have a huge chance with the new media manager which allows us for example to modify all the images in 16/9.
I don't like templates that cut images because often we don't know how to use them. But that's just my opinion.

chmst commented:
I have tested this item :white_check_mark: successfully on cddad60a16a6c7694ec5a23e79423547b47362ad<br /><br />I have tested this successfully. As it is described, it resolves the issue that the horizontal alignment was not possible. The styling will be a second step and is work in progress. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30527">issues.joomla.org/tracker/joomla-cms/30527</a>.</sub>

hans2103 commented:
I have tested this item :white_check_mark: successfully on cddad60a16a6c7694ec5a23e79423547b47362ad<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30527">issues.joomla.org/tracker/joomla-cms/30527</a>.</sub>

infograf768 commented:
Some thoughts.
Should'nt there be a default css for this module?
It would be overridden by any 3rd party template but Cassiopea would use it.

In this case the css could be, or included in a css in the module folder, or an instance created in media/mod_articles_news/css/template.css
This last one through an xml `<media destination="mod_articles_news" etc.`
or as it is a core module, added to build/

This would mean that the cassiopea template css would not be crowded with specific modules css when they are not used.

drmenzelit committed:
Merge branch '4.0-dev' into 4.0-dev-newsflash-horiz

drmenzelit committed:
Moved css for mod_articles_news to media folder

User webAsset Manager to load specific css for module

drmenzelit commented:
I changed the mod_articles_news to use specific css independently from template. The css for mod_articles_news is now under media/mod_articles_news/css (after running npm ci).
I removed the cropping of the images, that can be added on the template if necessary.
I hope my commit is correct, I had some issues with my internet connection the last days :-)

drmenzelit commented:
Thank you @infograf768 for the hint with mod_languages :-)

infograf768 commented:
> Do you mean with "would not be crowded" the place where it is saved or if it is loaded?

Loaded.

infograf768 commented:
> I think css/scss is the V in MVC and should mainly be the template.

Agree but "mainly" is the key here. If we have a parameter for a core module, I think it should be taken care of by default in core and obviously allow to be overridden by any template, which is easy as the css is loaded by the module tmpl only when used.

This is the case for mod_languages, plg_editors_codemirror, plg_system_highlight, etc. Some using scss instead of css (why this difference btw? Not sure).

infograf768 commented:
@drmenzelit 
Concerning
```
$wa = $app->getDocument()->getWebAssetManager();
$wa->registerAndUseStyle('mod_modules', 'mod_articles_news/template.css');
```

I has to place this in `/modules/mod_articles_news/tmpl/horizontal.php` for the css to load when param is set to horizontal.

Also, I had to change minmax to 240px to get 3 columns instead of 2 here when module is displayed in main-top position and cassiopea is set to static. I understand that maybe voluntary.

I'm a bit confused between use of grid and flex, but that's certainly just me. ;)

drmenzelit commented:
@infograf768 
The question is if we really need to have a css file for horizontal and one for vertical, or if it is ok to have only one file with all necessary standard definition for the module.

Abot flex and grid, yes, sometimes I'm also confused ;-) The advantage of grid with auto-fit and minmax is that you don't need media queries. But it can be challenging to find the correct size.

infograf768 commented:
No idea if we need one css for vertical, but for sure here your new css have to be loaded in horizontal.php as they don't when loaded through default.php. Maybe one css only (including vertical and horizontal) needs to be made if needed for vertical, but it should be loaded correctly in each tmpl

drmenzelit committed:
Move web asset call to horizontal.php

drmenzelit commented:
@infograf768 yes, you are right! Corrected with the last commit

chmst commented:
I have tested this item :white_check_mark: successfully on daaaae4140e44f46d9417bdf14a35fb0c345a9b0<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30527">issues.joomla.org/tracker/joomla-cms/30527</a>.</sub>

Quy commented:
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/30527">issues.joomla.org/tracker/joomla-cms/30527</a>.</sub>

infograf768 commented:
Tks
